File: exe11.py
from abc import ABC, abstractmethod

class Funcionario(ABC):
    def __init__(self, nome):
        self.nome = nome

    @abstractmethod
    def calcular_salario(self):
        pass

class FuncionarioHorista(Funcionario):
    def __init__(self, nome, valor_por_hora, horas_trabalhadas):
        super().__init__(nome)
        self.valor_por_hora = valor_por_hora
        self.horas_trabalhadas = horas_trabalhadas

    def calcular_salario(self):
        return self.valor_por_hora * self.horas_trabalhadas

class FuncionarioAssalariado(Funcionario):
    def __init__(self, nome, salario_mensal):
        super().__init__(nome)
        self.salario_mensal = salario_mensal

    def calcular_salario(self):
        return self.salario_mensal

if __name__ == "__main__":
    f1 = FuncionarioHorista("Carlos", 20.0, 160)
    f2 = FuncionarioAssalariado("Ana", 3000.0)

    print("Salário de {}: {}".format(f1.nome, f1.calcular_salario()))
    print("Salário de {}: {}".format(f2.nome, f2.calcular_salario()))

File: exe11.java
abstract class Funcionario {
    protected String nome;

    public Funcionario(String nome) {
        this.nome = nome;
    }

    public abstract double calcularSalario();
}

class FuncionarioHorista extends Funcionario {
    private double valorPorHora;
    private int horasTrabalhadas;

    public FuncionarioHorista(String nome, double valorPorHora, int horasTrabalhadas) {
        super(nome);
        this.valorPorHora = valorPorHora;
        this.horasTrabalhadas = horasTrabalhadas;
    }

    @Override
    public double calcularSalario() {
        return valorPorHora * horasTrabalhadas;
    }
}

class FuncionarioAssalariado extends Funcionario {
    private double salarioMensal;

    public FuncionarioAssalariado(String nome, double salarioMensal) {
        super(nome);
        this.salarioMensal = salarioMensal;
    }

    @Override
    public double calcularSalario() {
        return salarioMensal;
    }
}

public class exe11 {
    public static void main(String[] args) {
        Funcionario f1 = new FuncionarioHorista("Carlos", 20.0, 160);
        Funcionario f2 = new FuncionarioAssalariado("Ana", 3000.0);

        System.out.println("Salário de " + f1.nome + ": " + f1.calcularSalario());
        System.out.println("Salário de " + f2.nome + ": " + f2.calcularSalario());
    }
}
